### PR TITLE
Updates to React 16.8.2 and Emotion 10

### PR DESCRIPTION
**Pull request by zacjones93**
### Changes made 
- Simplify directory structure.
- Yarn, workspaces, build scripts, changelog and readme.
- Update emotion `styled`  import from `react-emotion` to `@emotion/styled`
- Update emotion `keyframes` from `react-emotion` to `@emotion/core`
- Update `react#16.7.0-alpha.0->16.8.2`.
- Update `react-dom#16.7.0-alpha.0->16.8.2`.
- Update `emotion#9.2.12->10.0.7`.

### Notes
The update went really smooth. I added yarn workspaces as that makes it really easy to start each lesson and all the dependencies are shared so you only need to install once.

## 03 - The Trouble Lesson
Lesson 03, `add-linting-to-help-enforce-rules-when-using-react-hooks`, was really the only lesson that needs adjustment as far as I can tell apart from the changes that you mentioned about `useReducer` (I think 🤔). 

From the students perspective, it's really hard to tell whats going on with the `package.json` file. Are all those packages needed to make the linting work?

It would also be worth considering breaking the `eslintConfig` section out into it's own file to separate what is most important to the lesson from the rest of the `package.json` file.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Changelog
```
Comment (zacjones93) [inline CHANGELOG.md:1]: Added a changelog to be more explicit with the state of the course example

Comment (elijahmanor) [reply CHANGELOG.md:1]: Ohh that's a great idea!

**Comment by elijahmanor** [comment]
Hmm, so I've not used yarn much and definitely not workspaces. 

I know you removed the react-hooks folder level, but I'm seeing it still in the workspace part. Is that still needed?

Also, will this new structure still work with codesandbox? Can we point sandbox to a lesson and it'll be smart enough to pull the deps if they are in workspaces like this?

**Comment by zacjones93** [comment]
The yarn workspaces won't affect anything on codesandbox, the workspace is soley for running and updating the apps locally.

Each individual lesson was updated 'per normal' and still has a `package.json` which is what codesandbox uses to pull in dependencies. Essentially, codesandbox doesn't know about the workspace set up because we are pointing it at `/packages/{lesson-example}`.

Here's an example of a codesandbox example with the new structure: https://codesandbox.io/s/github/eggheadio-projects/egghead-course-react-hooks/tree/omega/packages/12-convert-a-render-props-using-a-custom-react-hook

> I know you removed the react-hooks folder level, but I'm seeing it still in the workspace part. Is that still needed?
@elijahmanor I'm not sure what you are referencing here with 'the workspace part', could you paste a link?

**Comment by elijahmanor** [comment]
Okay, cool. I just wanted to make sure before I merged this in. I didn't want to accidentally break all the embedded codesandboxes. Thanks!

**Pull request merged**